docs/docs/vm-fundamentals/intro.md

```diff
@@ -2,6 +2,73 @@
+
+## Cairo Runner
+
+The Cairo Runner serves as the practical interface between theoretical computational models and their execution within the Cairo framework because it is a pivotal tool in the transformation of abstract statements into verifiable computations, exemplified by converting a claim such as "the j-th Fibonacci number is y" into inputs that the deterministic and nondeterministic Cairo machines can process. The success of these inputs in the respective machines substantiates the truth of the original statement, leveraging the Cairo AIR and the STARK protocol to demonstrate the validity of these computations.
+
+As a specialized computer program, the Cairo Runner is responsible for executing compiled Cairo programs, which notably include nondeterministic code, this execution is distinct from that of conventional programs due to Cairo's allowance for nondeterminism. For example, in a scenario where the computation involves finding the square root of a number, the Cairo Runner facilitates the execution by considering both possible solutions (e.g., +Y and -Y) and selecting the one that satisfies the subsequent program instructions.
+
+This process is supported by "hints," which are special instructions that the Cairo Runner utilizes to resolve nondeterministic outcomes that cannot be directly inferred. These hints, typically written in Python, are crucial for the execution of certain Cairo programs where additional information is necessary to determine the course of execution.
+
+## Resources
+
+In the Cairo Virtual Machine, state transitions, influenced by each instruction, play a crucial role in the VM's operation. Unlike traditional deterministic CPU architectures, Cairo can handle nondeterministic transitions, allowing it to verify a wider range of computational statements efficiently. However, if a valid subsequent state cannot be established leading to what's known as an "execution rejection" the prover cannot generate proof for that program's execution. Therefore, this highlights the importance of stringent resource management within Cairo, ensuring that the VM operates within its limits and maintains the overall stability and security of the Starknet platform.
+
+Managing resources effectively is essential for preventing scenarios where an execution might be rejected due to the system hitting resource limits or encountering computationally infeasible states. By carefully monitoring and constraining the use of memory, computational steps, and data outputs, the Cairo VM ensures that each program can be executed and verified without risking the overall stability and security of the Starknet platform.
```

Comment:
All this is not accurate.

The CASM bytecode by construction will never fail. It is not possible to compile cairo 0 or cairo 1 code that can fail

See
https://medium.com/nethermind-eth/under-the-hood-of-cairo-1-0-exploring-sierra-7f32808421f5

This means the Cairo VM is just here tu run CASM bytecode that we know will not fail.

> By carefully monitoring and constraining the use of memory, computational steps, and data outputs, the Cairo VM ensures that each program can be executed and verified without risking the overall stability and security of the Starknet platform.

The cairo VM just executes Casm bytecode,